http-client/Network/HTTP/Client/Types.hs

```diff
@@ -147,12 +151,14 @@ data HttpExceptionContent
                    --
                    -- @since 0.5.0
                    | OverlongHeaders
-                   -- ^ Either too many headers, or too many total bytes in a
-                   -- single header, were returned by the server, and the
-                   -- memory exhaustion protection in this library has kicked
-                   -- in.
+                   -- ^ Too many total bytes in a single header field were
+                   -- returned by the server.
                    --
                    -- @since 0.5.0
+                   | TooManyHeaders
```

Comment:
The introduction of this error might be considered a breaking change, but it also simplified debugging the code.

Reply:
This is acceptable, we've documented that adding extra constructors to error types is something end users should expect.